pjmedia/src/pjmedia/stream.c

```diff
@@ -1324,16 +1332,30 @@ static pj_status_t put_frame_imp( pjmedia_port *port,
      */
     if (stream->use_ka)
     {
-	pj_uint32_t dtx_duration;
+        if (stream->start_ka_count) {
+            pj_timestamp now;
+            unsigned elapse_time;
 
-	dtx_duration = pj_timestamp_diff32(&stream->last_frm_ts_sent,
-					   &frame->timestamp);
-	if (dtx_duration >
-	    PJMEDIA_STREAM_KA_INTERVAL * PJMEDIA_PIA_SRATE(&stream->port.info))
-	{
-	    send_keep_alive_packet(stream);
-	    stream->last_frm_ts_sent = frame->timestamp;
-	}
+            pj_get_timestamp(&now);
+            elapse_time = pj_elapsed_msec(&stream->last_start_ka_tx, &now);
+	    if ((elapse_time > stream->start_ka_interval))
+	    {
+	        send_keep_alive_packet(stream);
+	        stream->last_start_ka_tx = now;
+                stream->start_ka_count--;
+	    }
+        } else {
```

Comment:
If outgoing RTP has been started, is there a chance that KA & RTP packets will be sent at the same time? may it cause any issue (e.g: remote jitter buffer may overwrite audio payload with KA/empty payload)?
Anyway, perhaps it can be minimized if the start-KA also consider `last_frm_ts_sent`.

Reply:
use ```last_frm_ts_sent``` into interval calculation